proto/invocation.proto

```diff
@@ -532,3 +541,75 @@ message GetTrendResponse {
 
   bool has_invocation_stat_percentiles = 4;
 }
+
+enum Metric {
+  UNKNOWN_METRIC = 0;
+  DURATION_USEC_METRIC = 1;
+  CAS_CACHE_MISSES_METRIC = 2;
+  EXECUTION_QUEUE_TIME_USEC_METRIC = 3;
+}
+
+message StatFilter {
+  Metric metric = 1;
+  int64 min = 2;
+  int64 max = 3;
+}
+
+// Fetches a heatmap for the requested metric--a heatmap is basically a set of
+// columns, where each column represents a histogram for that metric in a window
+// of time.
+message GetStatHeatmapRequest {
+  context.RequestContext request_context = 1;
+  TrendQuery query = 2;
+  Metric metric = 3;
+  int64 num_metric_buckets = 4;
+}
+
+message HeatmapColumn {
+  string timestamp = 1;
```

Comment:
why not an int64?

Reply:
I was playing along with other trends RPCs, and using dates makes it easier to write the query, but being able to bucket into windows smaller than days would be nice and it seems like we have everything in place to just use an int64 here, so sure.